arangod/Graph/Providers/ClusterProvider.cpp

```diff
@@ -416,8 +448,15 @@ auto ClusterProvider<StepImpl>::expand(
 
   if (ADB_LIKELY(relations != _vertexConnectedEdges.end())) {
     for (auto const& relation : relations->second) {
-      bool const fetched = _vertexConnectedEdges.contains(relation.second);
-      callback(Step{relation.second, relation.first, previous, fetched});
+      bool const fetchedTargetVertex =
+          _vertexConnectedEdges.contains(relation.second);
+      // [GraphRefactor] TODO: KShortestPaths does not require Depth/Weight. We
```

Comment:
I would hide this within the Step, a step should claim if it needs a weight or not, if it needs one it should reserve space otherwise not, and then this call here should be oblivious (or constexpr if) to this knowledge.
We hand in the Step as template anyways, so we can make Weighted and NonWeighted Steps.
But that is for later!

Reply:
Ticket created: GORDO-1366